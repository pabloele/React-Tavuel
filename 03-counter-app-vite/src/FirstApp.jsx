import PropTypes from "prop-types";

function FirstApp({ title, subtitle, name }) {
  if (!title) throw new Error("El title no existe");
  return (
    <>
      <div data-testid="test-title">TITULO</div>
      <h1>{title}</h1>
      <p>{subtitle + 1}</p>
      <p>{name}</p>
    </>
  );
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
  title: "No hay t√≠tulo",
  subtitle: 0,
  name: "Pablo",
};

export default FirstApp;
