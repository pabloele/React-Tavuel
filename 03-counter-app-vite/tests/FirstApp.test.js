import { getByTestId, render } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe("Pruebas en <FirstApp />", () => {
  test("Debe hacer match con el snapshot ", () => {
    const title = "Hola soy Juan Perez";
    const { container, getByText, getByTestId } = render(
      <FirstApp title={title} subtitle={1234} />
    );
    // console.log(container);
    expect(container).toMatchSnapshot();
    // expect(getByText(title)).toBeTruthy();
    // const h1 = container.querySelector("h1");
    // console.log(h1.innerHTML);
    // expect(h1.innerHTML).toBe(title);
    // expect(h1.innerHTML).toContain(title);
    // expect(getByTestId("test-title")).toBeTruthy;
    // expect(getByTestId("test-title").innerHTML).toBe(title);
  });
  test("Debe mostrar el subtÃ­tulo enviado por props ", () => {
    const title = "Hola soy Juan Perez";
    const subtitle = 1234;
    const { container, getByText, getByTestId, getAllByText } = render(
      <FirstApp title={title} subtitle={subtitle} />
    );

    expect(getByText(subtitle + 1)).toBeTruthy();
    // expect(getAllByText(title).length).toBe(2);
  });
});
