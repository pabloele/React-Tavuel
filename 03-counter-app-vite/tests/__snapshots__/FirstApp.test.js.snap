// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot  1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola soy Juan Perez
  </div>
  <h1>
    Hola soy Juan Perez
  </h1>
  <p>
    1235
  </p>
  <p>
    Pablo
  </p>
</div>
`;
